xarray/backends/h5netcdf_.py

```
@@ -100,7 +102,12 @@ def prepare_variable(self, name, variable, check_encoding=False):
         if dtype is str:
             dtype = h5py.special_dtype(vlen=unicode_type)
 
-        self.set_necessary_dimensions(variable)
+        unlimited_dims = self.encoding.get('unlimited_dims', set())
+        if len(unlimited_dims) > 0:
+            warnings.warn('h5netcdf does not support unlimited dimensions',
```

If `check_encoding` is `True`, this should raise an error, not just a warning.